ctenex/bot/bots/alpha/exchange_bot.py

```diff
@@ -0,0 +1,299 @@
+                    )
+                    """
+                ),
+                price_moments,
+            )
+            await session.commit()
+
+        # TODO: Update strategy. Issue #22
+
+    async def run(self) -> None:
+        logger.info(f"Starting exchange bot for contract {self.contract_id}")
+
+        current_timestamp = datetime.now(timezone.utc)
+        start_timestamp = current_timestamp - timedelta(
+            milliseconds=int(self.base_drift_in_ms)
+        )
+        end_timestamp = start_timestamp + timedelta(
+            milliseconds=int(self.sample_interval_in_ms)
+        )
+
+        logger.debug(
+            f"Getting orders for interval [{start_timestamp} - {end_timestamp}]"
+        )
+        orders_in_exchange = await self.get_orders(
+            contract_id=self.contract_id,
+            start_time=start_timestamp,
+            end_time=end_timestamp,
+        )
+
+        while True:
+            await self.process_orders(orders=orders_in_exchange, session_stream=db())
+
+            current_timestamp = datetime.now(timezone.utc)
+
+            start_timestamp = end_timestamp
+            end_timestamp = start_timestamp + timedelta(
+                milliseconds=int(self.sample_interval_in_ms)
+            )
+            drift_error_in_ms = (current_timestamp - start_timestamp) / timedelta(
+                milliseconds=1
+            )
+
+            ### Use the error to adjust the period and keep the drift constant
+            real_drift_in_ms = int(self.base_drift_in_ms) - drift_error_in_ms
+            adjusted_period_in_millis = real_drift_in_ms - drift_error_in_ms
+            # logger.debug(
+            #     f"Real drift in ms: {real_drift_in_ms}. "
+            #     f"Drift error in ms: {drift_error_in_ms}. "
+            #     f"Adjusted period in ms: {adjusted_period_in_millis}"
+            # )
+            await asyncio.sleep(adjusted_period_in_millis / 1000)
+
+            logger.debug(
+                f"Getting orders for interval [{start_timestamp} - {end_timestamp}]"
+            )
+            orders_in_exchange = await self.get_orders(
+                contract_id=self.contract_id,
+                start_time=start_timestamp,
+                end_time=end_timestamp,
+            )
+
+    async def close(self) -> None:
+        await self.exchange_client.aclose()
+
+
+def min_and_max_price_for_limit_orders(
+    orders: list[OrderGetResponse],
+) -> tuple[Decimal, Decimal]:
+    """
+    Calculate the minimum and maximum price from the limit orders.
+    """
+
+    limit_bids = [
+        order.price
+        for order in orders
+        if order.type == "limit" and order.side == "buy" and order.price is not None
+    ]
+    limit_asks = [
+        order.price
+        for order in orders
+        if order.type == "limit" and order.side == "sell" and order.price is not None
+    ]
+
+    min_limit_bid = min(limit_bids) if limit_bids else Decimal(0.00)
```

Comment:
Hm, you later assign these to `best_bed` and `best_ask`...  Shouldn't these be the other way around? As in: max(limit_bids) and min(limit_asks) ?